R/mclosest.R

```diff
@@ -0,0 +1,85 @@
+#'@title Extract closest values in a pairwise manner between two matrices
+#'
+#' @description
+#'
+#' `mclosest` functions calculate the closest rows in a pairwise manner.
+#' It returns the index of the closest row in 'table' for each row in 'x'. 
+#'
+#' @param x `numeric`  matrix or data frame representing the query data. each 
+#' row will be compared to 'table'. Need to have the same numbers of rows and in 
+#' the same order as 'table'.
+#'
+#' @param table `numeric` matrix or data frame containing the reference data to 
+#' be matched with each row of 'x'. Need to have the same numbers of rows and in 
+#' the same order as 'x'.
+#'
+#' @param ppm `numeric` representing a relative, value-specific 
+#' parts-per-million (PPM) tolerance that is added to tolerance (default is 0). 
+#'
+#' @param tolerance `numeric` accepted tolerance.
+#'
+#' @return `numeric` vector of indices indicating the closest row of 'table' for
+#' each row of 'x'.
+#' 
+#' @details
+#' If, for a row of 'x', 2 rows of 'table' are closest only the index of first 
+#' row will be returned. 
+#' 
+#' For both the 'tolerance' and 'ppm' arguments, if their length is different to 
+#' the number of rows of 'x' and 'table', the input argument will be replicated 
+#' to match it. 
+#' 
+#'
+#' @author Philippine Louail
+#' 
+#' @importFrom MsCoreUtils ppm
+#'
+#' @export 
+#'
+#' @examples
+#'x <- data.frame(a = 1:5, b = 3:7)
+#'table <- data.frame(c = c(11, 23, 3, 5, 1), d = c(32:35, 45))
+#'ppm <- 0.5
+#'tolerance <- 25
+#'
```

Comment:
you should actually run the example. I would suggest you replace the `ppm <- 0.5` and `tolerance <- 25` with `mclosest(x, table)` to show how results look like if the closest, regardless of the difference, is returned and once also specifying a value for `tolerance` to return for some rows a `NA`. Also, you can use `##` to add comments. So, it could look like:

```r
#' ## Get for each row of `x` the index of the row in `table` with the smallest difference 
#' ## of values (per column)
#' mclosest(x, table)
#'
#' ## If the absolute difference is larger than `tolerance`, return `NA`. Note
#' ## that the tolerance value of `25` is used for difference for each pairwise 
#' ## column in `x` and `table`.
#' mclosest(x, table, tolerance = 25)
```

Ah, and you'll need to put a whitespace between the `#'` and the e.g. `x <- data...`.

Reply:
I think it is fixed now, thanks a lot .